### DISCUSSION TITLE
get Feature by SearchHit in Java

### DISCUSSION THREAD

**Discussion by Meibes** [flatgeobuf/flatgeobuf#148] [Q&A]
Hello,

I'm trying to use FlatGeobuf in an Android-App using the SAF-Framework. The SAF-Frameworks only allows InputStreams and no filepaths afaik, that's why I can't use librarys like GDAL unless I copy the FlatGeobuf to a place where I can use filepaths (which I only want to do as a last resort).

By looking through the tests, I found the [PackedRTreeTest](https://github.com/flatgeobuf/flatgeobuf/blob/master/src/java/src/test/java/org/wololo/flatgeobuf/test/PackedRTreeTest.java), which nearly does everything I need: it opens a FlatGeobuf, filters features by a boundingbox (as far as I understand) and returns the index and offset of the filtered items. The only thing that is missing is how to convert the SearchHit with it's index and offset to an actual Feature with Geometry, so I can access all attributes and coordinates.

greetings

**Comment by bjornharrtell** [comment]
Yeah for Java this part of the reference implementation is still a TODO. Part of what is needed exists as GeoTools specific logic at https://github.com/geotools/geotools/tree/main/modules/unsupported/flatgeobuf. But reference implementation should not depend on GeoTools which means a basic Feature class will be needed.

Contributions or sponsoring for this feature are welcome.
